apps/arweave/src/ar_data_sync_worker.erl

```diff
@@ -82,7 +89,9 @@ read_range({_Start, _End, _OriginStoreID, TargetStoreID, _SkipSmall} = Args) ->
 		false ->
 			case ar_data_sync:is_disk_space_sufficient(TargetStoreID) of
 				true ->
-					read_range2(Args);
+					?LOG_DEBUG([{event, read_range},
+						{size, (_End - _Start) / (1024*1024)}, {args, Args}]),
+					read_range2(?READ_RANGE_MESSAGES_PER_BATCH, Args);
```

Comment:
Limit the number of `read_range` messages that will be sent at once. This avoids an OOM exception that can occur when handling ranges that contains thousands of small chunks (e.g. 4-byte chunks that each get packed up to 256KiB)